tests/transport/test_cache.py

```diff
@@ -0,0 +1,199 @@
+import datetime as dt
+import time
+from unittest.mock import Mock
+
+import pytest
+
+from tradingstrategy.timebucket import TimeBucket
+
+
+@pytest.fixture()
+def class_under_test():
+    from tradingstrategy.transport.cache import CachedHTTPTransport
+
+    return CachedHTTPTransport
```

Comment:
Personally I would create the mock transport fixture directly, as the fixture name is clearer and less code to write down the line:

```
# i guess it's fine to move this import to module level as we are not trying to mock anything there
from tradingstrategy.transport.cache import CachedHTTPTransport


@pytest.fixture()
def transport():
    return CachedHTTPTransport(download_func=Mock())
```

Reply:
Makes sense, in the end it turned out that instance creation does not need to be customized and the tests repeatedly do the very same instantiation.

The reason for having the import of the thing-under-test is that if something gets messed up, we still want to see failing tests instead of an error that prevents the test suite from running at all. It's a pattern inspired to what Pyramid folks do, thus I prefer keeping the import where it already is.